### PR TITLE
feat(dataframe): support column and row Series

### PR DESCRIPTION
**Pull request by AChenQ**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2b4rn1u 实现 Series 类，支持通过 index 和 integer-location 索引数据](https://app.clickup.com/t/2b4rn1u)

File: graviti/dataframe/column/series.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    def __getitem__(self, key: Union[int, Iterable[int]]) -> Any:
+        integer_location = self._get_loc_by_index(key)
+
+        if isinstance(integer_location, list):
+            new_data = [self._data[loc] for loc in integer_location]
+            return Series(new_data, name=self.name, index=integer_location)
+
+        return self._data[integer_location]
+
+    def _get_loc_by_index(self, key: Union[int, Iterable[int]]) -> Union[int, List[int]]:
```
Comment (linjiX) [inline graviti/dataframe/column/series.py:80]: shall we add overload for `_get_loc_by_index`?

File: graviti/dataframe/column/series.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        if self._indices_data is None:
+            if isinstance(key, Iterable):
+                return [index for index in key]  # pylint: disable=unnecessary-comprehension
```
Comment (linjiX) [inline graviti/dataframe/column/series.py:83]: `list(key)`?

File: graviti/dataframe/column/series.py (hunk 1 above)
Comment (linjiX) [reply graviti/dataframe/column/series.py:80]: What is the difference between `loc` and `location`? why we have `loc` and `location` in function name?

File: graviti/dataframe/column/series.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    @overload
+    def _getitem_by_location(self, key: Iterable[int]) -> "Series":
+        ...
+
+    def _getitem_by_location(self, key: Union[int, Iterable[int]]) -> Union["Series", Any]:
+        if isinstance(key, int):
+            return self._data[key]
+
+        if self._indices is None:
+            index = [idx for idx in key]  # pylint: disable=unnecessary-comprehension
```
Comment (linjiX) [inline graviti/dataframe/column/series.py:109]: `list(key)`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/column/series.py (hunk 3 above)
Comment (linjiX) [reply graviti/dataframe/column/series.py:109]: `indices`?

Comment (linjiX) [reply graviti/dataframe/column/series.py:109]: idx is abbreviation of index, it is weird idx and index appears at the same place

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/column/series.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    _indices_data: Optional[Dict[int, int]]
+    _indices: Optional[List[int]]
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        data: Sequence[Any],
+        schema: Any = None,
+        name: Optional[Union[str, int]] = None,
+        index: Optional[List[int]] = None,
```
Comment (linjiX) [inline graviti/dataframe/column/series.py:46]: should the index type be `Iterable`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/column/series.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            return [self._indices_data[index] for index in key]
+
+        return self._indices_data[key]
+
+    # @overload
+    # def _getitem_by_location(self, key: slice) -> "Series":
+    #    ...
+
+    @overload
+    def _getitem_by_location(self, key: int) -> Union["Series", Any]:
```
Comment (Lee-000) [inline graviti/dataframe/column/series.py:109]: In what case would the return type be of `Series`? If it does return a `Series`, should the schema not only be of a single dtype? And why the return type is different than `__getitem__`?

File: graviti/dataframe/row/series.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+_T = TypeVar("_T", int, str)
+
+
+class Series(SeriesBase[_T]):
+    """One-dimensional array.
+
+    Arguments:
+        data: The data that needs to be stored in Series. Could be ndarray or Iterable.
+        schema: Data type to force. Only a single dtype is allowed. If None, will be
```
Comment (Lee-000) [inline graviti/dataframe/row/series.py:20]: Could `A single dtype` describe all the column types?

**Review by Lee-000 — CHANGES_REQUESTED** [review]

**Review by graczhual — APPROVED** [review]

File: graviti/dataframe/column/series.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        2 3
+        3 4
+        schema: int
+
+    """
+
+    _indices_data: Optional[Dict[int, int]]
+    _indices: Optional[List[int]]
+
+    def __init__(  # pylint: disable=too-many-arguments
```
Comment (linjiX) [inline graviti/dataframe/column/series.py:41]: we have only 5 arguments here, is this disable comments useless?

File: graviti/dataframe/column/series.py [hunk 8]
```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""The implementation of the Lazy Series."""
```
Comment (linjiX) [inline graviti/dataframe/column/series.py:6]: Why "Lazy"?

**Review by linjiX — APPROVED** [review]

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**